calculator.rb

```diff
@@ -0,0 +1,96 @@
+# prints title and operation options
+puts "Are you ready to Calculate!!!"
+puts "Operations:\nadd(+)\nsubtract(-)\nmultiply(*)\ndivide(/)\nexponent(^)\nmodulo(%)\n"
+# collects operation type from user
+print "Choose an operation: "
+operation = gets.chop.downcase
+# list of acceptible operations
+operators = ["+", "add", "-", "subtract", "*", "multiply", "/", "divide", "exponent", "^", "modulo", "%", "()"]
+
+# confirms if user entered a  valid operation
+until operators.include?(operation)
+  puts "Please enter a valid operation"
+  operation = gets.chop.downcase
+end
+
+# collects two numbers for calculation
+puts "What are the two numbers you want to use?"
+print "Enter a number: "
+value_1 = gets.chomp
+# confirms user has entered a valid number
+until value_1.to_s == value_1.to_i.to_s || value_1.to_f.to_s == value_1.to_s
+  print "Please enter a valid number: "
+  value_1 = gets.chomp
+end
+
+print "Enter a second number: "
+value_2 = gets.chomp
+# confirms user has entered a valid number
+until value_2.to_s == value_2.to_i.to_s || value_2.to_f.to_s == value_2.to_s
+  print "Please enter a valid number: "
+  value_2 = gets.chomp
+end
+
+# methods used
+def add(num1, num2)
```

Comment:
This is more of a style thing.  It's a little odd putting the method definitions here in the middle of the program flow.  I suggest moving method definitions to the top of the file.